src/main/java/frc/robot/commandgroups/ShootAndRun.java

```diff
@@ -0,0 +1,50 @@
+package frc.robot.commandgroups;
+
+import edu.wpi.first.math.geometry.Translation2d;
+import edu.wpi.first.math.kinematics.ChassisSpeeds;
+import edu.wpi.first.wpilibj2.command.ParallelCommandGroup;
+import frc.robot.Constants;
+import frc.robot.Robot;
+import frc.robot.subsystems.conveyor.Conveyor;
+import frc.robot.subsystems.drivetrain.SwerveDrive;
+import frc.robot.subsystems.drivetrain.commands.TurnToAngle;
+import frc.robot.subsystems.drivetrain.commands.TurnWhileRunning;
+import frc.robot.subsystems.flap.Flap;
+import frc.robot.subsystems.hood.Hood;
+import frc.robot.subsystems.shooter.Shooter;
+import frc.robot.utils.Utils;
+
+import java.util.function.DoubleSupplier;
+import java.util.function.Supplier;
+
+public class ShootAndRun extends ParallelCommandGroup {
+
+
+    public ShootAndRun(Shooter shooter, SwerveDrive swerveDrive, Hood hood, Conveyor conveyor, Flap flap, DoubleSupplier visionDistance, DoubleSupplier visionYaw) {

```

Comment:
The next step here would be figuring out how to incorporate this so that the driver can still drive the robot. The driver moves the robot xy and the code changes the angle and shoots.